src/robusta/core/playbooks/internal/discovery_events.py

```diff
@@ -17,3 +18,59 @@ def cluster_discovery_updates(event: KubernetesAnyChangeEvent):
                 namespace=event.obj.metadata.namespace,
             )
         )
+
+
+@action
+def event_history(event: ExecutionBaseEvent):
+    """
+    Creates findings for the all the past events for any object that has had at one point a warning event
+    """
+    reported_obj_history_list = []
+    warning_events = get_events_list(event_type="Warning")
+    for warning_event in warning_events.items:
+        warning_event_key = gen_object_key(warning_event.involvedObject)
+        if warning_event_key in reported_obj_history_list:
+            # if there were multiple warnings on the same object we dont want the history pulled multiple times
+            continue
+        finding = create_event_finding(warning_event)
+        events_table = get_resource_events_table(
+            "Resource events",
+            warning_event.involvedObject.kind,
+            warning_event.involvedObject.name,
+            warning_event.involvedObject.namespace
+        )
+        if events_table:
+            finding.add_enrichment([events_table])
+        event.add_finding(finding)
```

Comment:
not possible to send with no events since there will be at least one warning event to get to this line and the events table will contain that information

Reply:
yes, I know. But we still have that if statement.
If we have it, I think it makes more sense to have the add_finding under it as well

Reply:
kk